demo/build.rs

```diff
@@ -0,0 +1,15 @@
+use std::env;
+use std::path::PathBuf;
+
+pub fn main() {
+    // Add rustup to the rpath in order to properly link with the correct rustc version.
+    let rustup_home = env::var("RUSTUP_HOME").unwrap();
+    let toolchain = env::var("RUSTUP_TOOLCHAIN").unwrap();
+    let rustc_lib: PathBuf = [&rustup_home, "toolchains", &toolchain, "lib"]
+        .iter()
+        .collect();
+    println!(
+        "cargo:rustc-link-arg-bin=smir-demo=-Wl,-rpath,{}",
+        rustc_lib.display()
+    );
```

Comment:
Is this something all users need to do? Can we do anything on the rustc side to make this simpler?

Reply:
I think users have to do this if they want to be able to invoke their binaries directly. If you run the program under cargo or rustup run, it should configure the environment correctly for you.

Reply:
I.e.: if you remove this build file, the following still works:
```
$ cargo run -- --help
```
But if you run the binary directly:
```
$ ./target/debug/smir-demo --help
./target/debug/smir-demo: error while loading shared libraries: librustc_driver-a03ea465d8e03db1.so: cannot open shared object file: No such file or directory
```